paasta_tools/cli/cmds/get_deploygroup_history.py

```diff
@@ -0,0 +1,177 @@
+        '"all.main", "all.main,all.canary").'
+        "Use the flag -a or --all-deploy-groups",
+        default="",
+        required=False,
+    )
+    arg_deploy_group.completer = lazy_choices_completer(list_deploy_groups)  # type: ignore
+
+    list_parser.add_argument(
+        "-a",
+        "--all-deploy-groups",
+        help="View history of all deploy groups for the service",
+        action="store_true",
+        required=False,
+    )
+
+    list_parser.add_argument(
+        "-y",
+        "-d",
+        "--soa-dir",
+        dest="soa_dir",
+        metavar="SOA_DIR",
+        default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+
+    list_parser.set_defaults(command=paasta_get_deploygroup_history)
+
+
+def get_versions_for_service(
+    service: str, deploy_groups: Collection[str], soa_dir: str
+) -> Mapping[DeploymentVersion, Tuple[str, str]]:
+    """Returns a dictionary of 2-tuples of the form (timestamp, deploy_group) for each version tuple of (deploy sha, image_version)"""
+    if service is None:
+        return {}
+    git_url = get_git_url(service=service, soa_dir=soa_dir)
+    all_deploy_groups = list_deploy_groups(service=service, soa_dir=soa_dir)
+    deploy_groups, _ = validate_given_deploy_groups(all_deploy_groups, deploy_groups)
+    previously_deployed_versions: Dict[DeploymentVersion, Tuple[str, str]] = {}
+
+    for ref, sha in list_remote_refs(git_url).items():
+        regex_match = extract_tags(ref)
+        try:
+            deploy_group = regex_match["deploy_group"]
+            tstamp = regex_match["tstamp"]
+            image_version = regex_match["image_version"]
+        except KeyError:
+            pass
+        else:
+            if deploy_group in deploy_groups:
+                version = DeploymentVersion(sha=sha, image_version=image_version)
+                previously_deployed_versions[version] = (tstamp, deploy_group)
+    return previously_deployed_versions
+
+
+def list_previous_versions(
+    service: str,
+    deploy_groups: Collection[str],
+    any_given_deploy_groups: bool,
+    versions: Mapping[DeploymentVersion, Tuple],
+) -> None:
+    def format_timestamp(tstamp: str) -> str:
+        return naturaltime(datetime_from_utc_to_local(parse_timestamp(tstamp)))
+
+    print("Below is a list of deploy_groups History:")
+    # All versions sorted by deployment time
+    list_of_versions = sorted(versions.items(), key=lambda x: x[1], reverse=True)
+    rows = [("Timestamp -- UTC", "Human time", "deploy_group", "Version")]
+    for version, (timestamp, deploy_group) in list_of_versions:
+        rows.extend(
+            [(timestamp, format_timestamp(timestamp), deploy_group, repr(version))]
+        )
+    for line in format_table(rows):
+        print(line)
+
+
+def paasta_get_deploygroup_history(args: argparse.Namespace) -> int:
+    soa_dir = args.soa_dir
+    service = figure_out_service_name(args, soa_dir)
```

Comment:
imo, it'd be better to directly call the `validate_service_name function` rather than call `figure_out_service_name` to validate - not to mention that `figure_out_service_name` will also call `exit(1)` for us if the service name is invalid (and the `guess_service_name` code in `figure_out_service_name` won't get used since `service` is a required arg)